test/lib/users/dialogs/inst_user_first.rb

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "../../../test_helper"
+require "users/dialogs/inst_user_first"
+
+describe Yast::InstUserFirstDialog do
+  subject(:dialog) { described_class.new }
+
+  describe "#choose_users_handler" do
+    let(:users_to_import_dialog) { instance_double(Yast::UsersToImportDialog, run: nil) }
+    let(:root2_path) { FIXTURES_PATH.join("root2") }
+    let(:root3_path) { FIXTURES_PATH.join("root3") }
+
+    context "when there is another Linux installation" do
+      before do
+        Users::UsersDatabase.import(root2_path)
+      end
+
+      it "displays a dialog with importable users found" do
+        expect(Yast::UsersToImportDialog).to receive(:new)
+          .with(["b_user", "c_user"], [])
+          .and_return(users_to_import_dialog)
+
+        dialog.choose_users_handler
+      end
+    end
+
+    context "when there are multiple Linux installations" do
+      before do
+        # Mock access time: files in root2 are more recent than files in root3
+        allow(File).to receive(:atime) do |path|
+          path =~ /root2/ ? Time.now : Time.now - 1200
```

Comment:
I am not sure if this is robust enough. I would use a specific time object:

```suggestion
          path =~ /root2/ ? Time.new(2018) : Time.new(2017)
```
Moreover, I would mock it in a different way:

```
allow(File).to receive(:atime).with(/root2/).and_return(Time.new(2018))
allow(File).to receive(:atime).with(/root3/).and_return(Time.new(2017))
```

Reply:
I must admit that this pieces of code were directly copy/paste from the existing one, without pay too much attention. But I agree with you, thank you Iván.